### DISCUSSION TITLE
Time offset for scheduling output on `TimeInterval`

### DISCUSSION THREAD

**Discussion by ammtD** [CliMA/Oceananigans.jl#2686] [Computational science!] [answered]
Hello, 

I would like to save average fields every x-seconds.  The averages will be computed over a period T (T<=0.5x).   Then I would like to also store full fields every  x-seconds,  but staggered such that I have a full field in the middle of the period over which the average is computed.
e.g.   120s average stored at t= 600s, 1200s, 1800s  etc., fields stored at t = 540s, 1140s, 1740s.   
To achieve this I imagine I'd need to specify the time of the first output,  next to the interval and window.... I bet this is this possible, could someone please guide me how?

```
# using NetCDF output writer:
simulation.output_writers[:field_writer] = NetCDFOutputWriter(model, outputs, filename="WSf_fields_all.nc", 
    schedule=TimeInterval(600),  overwrite_existing = true)

simulation.output_writers[:avg_output] = NetCDFOutputWriter(model, outputs, filename="WSf_avgs_all.nc", 
    schedule=AveragedTimeInterval(600, window=120), overwrite_existing = true)
```

**Comment by glwagner** [comment, marked as answer]
For this application I think you can use `AveragedTimeInterval` as is for the averaged fields.

For the unaveraged, "centered" fields, there's a few options. One is to provide the second (undocumented) argument to `TimeInterval`, which allows you to offset the interval:

```julia
schedule = TimeInterval(600, -60) # assuming that the model starts at t=0.
```

You may also use the `SpecifiedTimes` schedule (for which you explicitly specify all the times where you'd like output).

If you find the "offset" method useful, we may want to document it (and perhaps clarify the naming of that second property, which is now called `previous_actuation_time`).

If you want to understand how this works in the code, see here:

https://github.com/CliMA/Oceananigans.jl/blob/2234f02afdb83815ffd1134033b25c09477ddeac/src/Utils/schedules.jl#L17-L54

**Reply by glwagner** [reply]
Also do you mind if I elaborate the title of this discussion? I think this might be useful to others.

**Reply by ammtD** [reply]
Thanks, I am going to try the offset option.  Feel free to change the title of the discussion to make it clearer for others what this is about.

**Reply by glwagner** [reply]
Let us know if it works!

**Reply by ammtD** [reply]
I wrote it up as follows:

```
avg_window = 180     # seconds
offset = -1* avg_window/2     
simulation.output_writers[:field_writer] = NetCDFOutputWriter(model, outputs, filename="WSf_fields_all.nc", 
    schedule=TimeInterval(600, offset),  overwrite_existing = true)

simulation.output_writers[:avg_output] = NetCDFOutputWriter(model, outputs, filename="WSf_avgs_all.nc", 
    schedule=AveragedTimeInterval(600, window=avg_window), overwrite_existing = true)
```

Fieldwriter writes output at t=0, 510, 110, 1710 s.... etc.,  Averagewriter at t=0, 600, 1200, 1800 s .... etc.,  so I believe it works!

**Reply by glwagner** [reply]
Nice! We should update the docstring to advertise this feature then!
